### DISCUSSION TITLE
Insiders publish workflow not run when I create a release

### DISCUSSION THREAD

**Discussion by PEZ** [squidfunk/mkdocs-material#3176] [Q&A]
I am trying to install Insiders using the docker option: https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#with-docker

I followed the steps, but when I create a release, the publish workflow is not run.

I am not sure how  I am supposed to create a release, but I just use the UI and select  the most recent tag, then click **Publish release**.

Here's my fork: https://github.com/PEZ/mkdocs-material-insiders

I am probably doing something wrong, but I can't figure out what that would be.

**Comment by squidfunk** [comment]
Possibly related to this regex:
https://github.com/squidfunk/mkdocs-material-insiders/blob/5956a3c713f0477467e0f18708f663ad7ab8f333/.github/workflows/publish.yml#L25

I think we should switch to the [`release`](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release) trigger. Care to test this one and submit a PR on the Insiders repo?

**Reply by PEZ** [reply]
Oh, yes.

**Reply by PEZ** [reply]
It seems the workflow is only triggered when I push a new tag and create a release from that.

**Reply by PEZ** [reply]
The PR works, with the added work of pushing a new tag... Not that it works for my particular fork, because

```
Error: Build and push Docker image
buildx failed with: error: invalid tag "ghcr.io/PEZ/mkdocs-material-insiders:latest": repository name must be lowercase
```

That's the first time I have got complaints about my Github name 😄

**Reply by ironicbadger** [reply]
I changed mine tonight from `IronicBadger` to `ironicbadger` because of this issue - no big deal really I suppose but it would be nice to see some kind of parsing of usernames to convert them to lowercase automatically. It's a docker limitation btw.

Also created an issue just now which would suggest that this isn't working.
https://github.com/squidfunk/mkdocs-material/issues/3187

**Reply by PEZ** [reply]
I'm not going to change my GitHub user name. 😀 It's a rather big deal to me. 

I might have a look at that issue though. Did it fail despite you having changed the name? It looked like all lowercase in the messages...

**Reply by squidfunk** [reply]
> I changed mine tonight from IronicBadger to ironicbadger because of this issue - no big deal really I suppose but it would be nice to see some kind of parsing of usernames to convert them to lowercase automatically. It's a docker limitation btw.

I can understand that this is totally annoying. It's either an issue with GitHub or Docker, so maybe ask them?

I don't get why the Insiders workflow isn't working for you. If you create a release, it will also ask you for creating a tag. GitHub's UI might not be the best (they changed it recently), but that's definitely the way to go. No manual pusshing. I'll look into it this weekend and try to reproduce it on another account of mine.

However, it might be related to the latest change on the workflow on the Insiders repository, which now doesn't seem to receive the tag, which is why none is present in the push command. We might need to change it back to trigger on tagged commits.

**Comment by PEZ** [comment]
> I can understand that this is totally annoying. It's either an issue with GitHub or Docker, so maybe ask them?

I don't think any of those are malfunctioning, so I doubt they would give us much attention. I sent a PR on Insiders that lowercases the repository name and with that I can also build docker images. 😄
